spec/debug/blocks.cr

```diff
@@ -0,0 +1,12 @@
+def f(v)
+  yield(v)
+end
+
+f("hello world") do |v|
```

Comment:
I recommend to replace it with 
```
["hello world"].each do |v|
...
```
as it will generate proper testing as in this IR debug output:
```
!549 = distinct !DILexicalBlock(scope: !542, file: !550, line: 1, column: 1)
!550 = !DIFile(filename: "blocks.cr", directory: "/Users/sergey/Projects/crystal/crystal/test")
!551 = !DIDerivedType(tag: DW_TAG_pointer_type, name: "String", baseType: !552, size: 64, align: 64)
!552 = !DICompositeType(tag: DW_TAG_structure_type, name: "String", line: 1, size: 128, align: 128, elements: !553)
!553 = !{!554, !555, !556}
!554 = !DIDerivedType(tag: DW_TAG_member, name: "bytesize", line: 1, baseType: !4, size: 32, align: 32, offset: 32)
!555 = !DIDerivedType(tag: DW_TAG_member, name: "length", line: 1, baseType: !4, size: 32, align: 32, offset: 64)
!556 = !DIDerivedType(tag: DW_TAG_member, name: "c", line: 1, baseType: !120, size: 8, align: 8, offset: 96)
!557 = !DILocalVariable(name: "a", scope: !549, file: !550, line: 6, type: !551)
!558 = !DILocalVariable(name: "b", scope: !549, file: !550, line: 7, type: !4, align: 32)
!559 = !DILocalVariable(name: "v", scope: !549, file: !550, line: 1, type: !551)
!560 = !DILocation(line: 83, column: 40, scope: !561)
!561 = distinct !DILexicalBlock(scope: !542, file: !562, line: 1, column: 1)
!562 = !DIFile(filename: "hasher.cr", directory: "/Users/sergey/Projects/crystal/crystal/src/crystal")
!563 = !DILocation(line: 42, column: 1, scope: !564)
!564 = distinct !DILexicalBlock(scope: !542, file: !565, line: 1, column: 1)
!565 = !DIFile(filename: "kernel.cr", directory: "/Users/sergey/Projects/crystal/crystal/src")
!566 = !DILocation(line: 27, column: 23, scope: !567)
!567 = distinct !DILexicalBlock(scope: !542, file: !568, line: 1, column: 1)
!568 = !DIFile(filename: "lib_pcre.cr", directory: "/Users/sergey/Projects/crystal/crystal/src/regex")
!569 = !DILocation(line: 28, column: 21, scope: !567)
```